lib/bing_translator.rb

```diff
@@ -143,6 +146,33 @@ def balance
   # Returns existing @access_token if we don't need a new token yet,
   # or returns the one just obtained.
   def get_access_token
+    if @subscription_key
+      get_cognitive_access_token
+    else
+      get_datamarket_access_token
+    end
+  end
+
+private
+
+  def get_cognitive_access_token
+    headers = {
+      'Ocp-Apim-Subscription-Key' => @subscription_key
+    }
+
+    http = Net::HTTP.new(@cognitive_token_uri.host, @cognitive_token_uri.port)
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE if @skip_ssl_verify
+
+    response = http.post(@cognitive_token_uri.path, "", headers)
+
+    @access_token = {
+      'access_token' => response.body,
+      'expires_at' => Time.now + 480
```

Comment:
@abrisse if I read the [documentation](https://msdn.microsoft.com/en-us/library/hh454950.aspx) correctly (and this seems to relate to the azure service), the response:
```
Property          Description
access_token  The access token that you can use to authenticate you access to the Microsoft Translator API.
token_type      The format of the access token. Currently, Azure DataMarket returns http://schemas.xmlsoap.org/ws/2009/11/swt-token-profile-1.0 , which indicates that a Simple Web Token (SWT) token will be returned.
expires_in        The number of seconds for which the access token is valid.
scope              The domain for which this token is valid. For the Microsoft Translator API, the domain is http://api.microsofttranslator.com.  
```
contains the property *expires_in* which can be used to calculate the expiration date.